server/src/main/java/com/networknt/server/DefaultConfigLoader.java

```diff
@@ -90,6 +93,8 @@
 
     // An instance of Jackson ObjectMapper that can be used anywhere else for Json.
     final static ObjectMapper mapper = new ObjectMapper();
+    // The instance of Http2Client that is used to connect to the light-config-server with bootstrap.truststore
+    static Http2Client client = Http2Client.getInstance();
```

Comment:
Hi Steve, unfortunately this doesn't solve the issue.

The config loader has a very different SSL context than the application it is configuring. The certs and private keys that the application needs are not available until AFTER the config loader pulls them from config server.

So, by the time the the application's certs have been downloaded by the config loader, it is already too late to incorporate these app certs into the SSL context, since the config loader has already created an SSL context.

Also: The original Apache http client was very bare-boned, in that it did not search for items such as client.yml, service.yml, server.yml, etc.... The solution you propose causes all the usual config files to be loaded.... but the problem is that these files are still on the config server. So by the time the application's certs and config files have been downloaded, it is too late, since the Http2Client has already created an SSL context that the application cannot use, and has loaded the default config files.

Reply:
Maybe I missed something but what I did is just replace the Apache HttpClient with Http2Client and didn't modify any flow. In both cases, you need to have the bootstrap certificate to be ready in order to connect to the config server. I am not very familiar with the flow and want to have a walkthrough session together to clarify it.